src/main/java/com/yahoo/bullet/querying/RunningQuery.java

```diff
@@ -57,7 +60,12 @@ public RunningQuery(String id, String queryString, BulletConfig config) {
     @Override
     public Optional<List<BulletError>> initialize() {
         start();
-        return query.initialize();
+        Optional<List<BulletError>> errors = query.initialize();
+        if (!errors.isPresent()) {
+            filter = new Filter(query.getFilter());
```

Comment:
It's weird that we only do this for the new expression stuff. Aggs and post aggs are in Querier. I feel like it should all be in Querier.

Reply:
Yea, that might work better. In Querier#initialize(), I can have 

filter = new Filter(runningQuery.getQuery().getFilter()); 
projection = new Projection(runningQuery.getQuery().getProjection());

after runningQuery#initialize()